internal/pkg/agent/install/install.go

```diff
@@ -301,3 +369,22 @@ func hasAllSSDs(block ghw.BlockInfo) bool {
 
 	return true
 }
+
+func createSocketDir(uid int, gid int) error {
```

Comment:
UID and GID are passed around so much it might be worth encapsulating them into a structure. At minimum that would let us initialize them by name to minimize the chance that someone accidentally reverses them.

Reply:
I like that sugession. Will make that change.

Reply:
If you're going this struct route (which I'm also ++ for), I'd consider encapsulating the string counterparts in the struct and adding accessor methods for returning the corresponding int values.  You could probably just do the conversion once, as part of the struct's constructor, not every time the accessors are called.  Currently, this string->int conversion is being done in a couple of places so moving this into a single place would be nice.

Reply:
Changing the the struct route allows me to define different structs between unix and Windows, which made the code a lot cleaner and much easier to understand.